app/sseclient/client.go

```diff
@@ -0,0 +1,214 @@
+// Copyright © 2022-2025 Obol Labs Inc. Licensed under the terms of a Business Source License 1.1
+
+package sseclient
+
+import (
+	"bufio"
+	"bytes"
+	"context"
+	"io"
+	"net/http"
+	"strconv"
+	"time"
+
+	"github.com/obolnetwork/charon/app/errors"
+	"github.com/obolnetwork/charon/app/z"
+)
+
+type Event struct {
+	ID    string
+	Event string
+	Data  []byte
+}
+
+type (
+	ErrorHandler func(err error, url string) error
+	EventHandler func(e *Event, url string) error
+)
+
+type Client struct {
+	URL        string
+	Retry      time.Duration
+	HTTPClient *http.Client
+	Headers    http.Header
+}
+
+var errStreamConn = errors.New("cannot connect to the stream")
+
+func New(url string) *Client {
+	return &Client{
+		URL:        url,
+		HTTPClient: &http.Client{},
+		Headers:    make(http.Header),
+	}
+}
+
+// Start connects to the SSE stream. This function will block until SSE stream is stopped.
+func (c *Client) Start(ctx context.Context, eventFn EventHandler, errorFn ErrorHandler) error {
+	timeout := c.Retry / 32
```

Comment:
Please check if `app/expbackoff` is sufficient to do the same in a nicer way.

Reply:
I forgot about our expbackoff! Will utilise it here.